packages/components/psammead-idt1-include/src/index.test.jsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
+import { render, wait } from '@testing-library/react';
+import Idt1Include from './index';
+import { readFileSync } from 'fs';
+
+const requireJsSrc = `https://news.files.bbci.co.uk/include/vjassets/js/vendor/require-2.1.20b.min.js`;
+const fakeInclude = readFileSync(`${__dirname}/fakeinclude.html`, 'utf8');
+
+describe('Idt1Include', () => {
+  shouldMatchSnapshot(
+    'should render with require js in head',
+    <Idt1Include html={fakeInclude} requireJsSrc={requireJsSrc} />,
+  );
+
+  it('should initialise the fake include utilising require js', async () => {
+    const { debug, getByText } = render(
+      <Idt1Include html={fakeInclude} requireJsSrc={requireJsSrc} />,
+    );
+    await wait(() => getByText('Initialised include'));
+  });
```

Comment:
This test does not work yet. 

In theory, the test should 
- Render the component, initially with no include in the body
- After a short amount of time require loads in the virtual browser
- The react component rerenders and the `fakeInclude` is added to the page
- The inline script from `fakeinclude.html` runs, uses require to modify the placeholder div, setting it's text to "Initialised include"
- `await wait(() => getByText('Initialised include'));` finishes its wait before the timeout, indicating the DOM has been updated by the inline script.

Either test is wrong, the implementation or both 😉  That's where I've got to 😓

Reply:
This _may_ be why I'm having issues: https://github.com/facebook/react/issues/8838#issuecomment-314740580